pkg/credentials/credential.go

```diff
@@ -0,0 +1,37 @@
+package credentials
+
+import (
+	"encoding/json"
+
+	"github.com/docker/cli/cli/config/types"
+)
+
+type Credential struct {
+	ToolID string            `json:"toolID"`
```

Comment:
Should this be toolname or tool? Not id.  FYI, ID's are not really "portable" They are basically generated on demand and not really guaranteed to be stable over time.  Only make sense in the context of a single "Program"

Reply:
Fixed and renamed to `ToolName`. Though based on a discussion that I had with Taylor earlier today, we are probably going to need to use something else as the identifier/unique name for each credential